src/Umbraco.Cms.Integrations.DAM.Aprimo/Services/AprimoService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Dynamic;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+
+using Umbraco.Cms.Integrations.DAM.Aprimo.Models;
+
+namespace Umbraco.Cms.Integrations.DAM.Aprimo.Services
+{
+    public class AprimoService : IAprimoService
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        private readonly OAuthConfigurationStorage _oauthConfigurationStorage;
+
+        public AprimoService(IHttpClientFactory httpClientFactory, OAuthConfigurationStorage oauthConfigurationStorage)
+        {
+            _httpClientFactory = httpClientFactory;
+
+            _oauthConfigurationStorage = oauthConfigurationStorage;
+        }
+
+        public async Task<AprimoResponse<Record>> SearchRecord(Guid id)
+        {
+            var client = _httpClientFactory.CreateClient(Constants.AprimoClient);
+
+            var oauthConfiguration = _oauthConfigurationStorage.Get();
+            if (oauthConfiguration == null) return AprimoResponse<Record>.Fail(Constants.ErrorResources.Unauthorized, false);
+
+            client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Umbraco.Cms.Integrations.DAM.Aprimo", "1.0.0"));
```

Comment:
Also, headers that are always the same can be set as `DefaultRequestHeaders` where the named client is setup, so they don't need to be set on every request.

Things that change per request should probably be on the headers of the request rather than the default headers of the client.

Reply:
Updated